### ISSUE TITLE
Linking errors when using web::AsyncWebrequest

### ISSUE THREAD

**Issue by Fleeym** [geode-sdk/geode#154]
### Platform

Windows

### SDK commit

3dd6d55

### Geode Version

v1.0.0-beta.13

### Mods Installed

- geode.devtools v1.1.0
- my own mod

### Expected Behavior

The mod compiles when using web::AsyncWebRequest

### Actual Behavior

I get two linking errors, one related to geode::utils::web::SentAsyncWebRequest::error, and another related to geode::utils::web::SentAsyncWebRequest::~SentAsyncWebRequest

- SentAsyncWebRequest::error happens when I use .then
- SentAsyncWebRequest::~SentAsyncWebRequest happens when i use .cancelled

![image](https://user-images.githubusercontent.com/61891787/230155463-7e3ee02e-f41b-4251-9130-ebc63006e698.png)

### Steps to Reproduce

1. Make a fetch request using web::AsyncWebRequest
2. .json()
3. implement .then, .expect and .cancelled

### Additional Information

Here is my code
```cpp
void fetchSFH(int songID, std::function<void(bool)> callback) {
        std::string url = "https://songfilehub.com/api/v1/nongs?id=" + std::to_string(songID);
        log::info("Request URL: {}", url);
        web::AsyncWebRequest()
            .fetch(url)
            .json()
            .then([callback, songID](json::Value const& data) {
                log::info("Entered callback");
                std::vector<SFHItem> ret;
                if (!data.contains("songs") || !data["songs"].is_array()) {
                    callback(false);
                }
                auto songs = data["songs"].as_array();
                for (auto const& song : songs) {
                    SFHItem item = {
                        .songName = song["songName"].as_string(),
                        .downloadUrl = song["downloadUrl"].as_string(),
                        .levelName = song.contains("levelNameMobile") ? song["levelNameMobile"].as_string() : "",
                        .songURL = song.contains("songURL") ? song["songURL"].as_string() : ""
                    };
                    ret.push_back(item);
                }
                addNongsFromSFH(ret, songID);
                callback(true);
            })
            .expect([callback](std::string const& error) {
                log::debug("oopsied");
                callback(false);
            })
            .cancelled([callback](auto r) {
                log::debug("oops");
                callback(false);
            });
        log::info("test");
    }
```

**Comment by altalk23** [comment]
Should be fixed by compiling against the binary of 5924901

**Comment by Fleeym** [comment]
Yep, it works, thanks
